builtin/logical/ad/backend_config_test.go

```diff
@@ -84,65 +80,64 @@ func TestConfigWriteReadDelete(t *testing.T) {
 	}
 }
 
-func configIsStoredAsExpected() bool {
+func configIsStoredAsExpected(t *testing.T) {
 	entry, err := storage.Get(ctx, config.StorageKey)
 	if err != nil {
-		return false
+		t.Error(err)
 	}
 
 	engineConf := &config.EngineConf{}
 	if err := entry.DecodeJSON(engineConf); err != nil {
-		return false
+		t.Error(err)
 	}
 
 	if engineConf.ADConf.Certificate != validCertificate {
-		return false
+		t.FailNow()
```

Comment:
I think generally we don't go for `t.FailNow()` and instead use `t.Fatal(...)` or `t.Fatalf(...)` to output more verbose failure messages. In this case, for example, it'd be helpful to output actual vs expected values. Same applies for the failure cases below. In cases where there's nothing to interpolate, a `t.Fatal(...)` should suffice.

Reply:
That's helpful. Will do.